public/scripts/upload.js

```diff
@@ -352,33 +429,52 @@ function updateFileListItem(itemNode, imageUpload) {
     isSameAspectRatio(imageDim, recommendDim) ||
     isSameAspectRatio(imageDimRotate, recommendDim);
   const shouldShowAspectRatioWarning =
-    imageUpload.readState !== ReadState.Reading && !isCorrectDim;
+    imageUpload.state !== ImageUploadState.Reading && !isCorrectDim;
   // Update status icon
   // error status has higher precedence over warning
   if (imageUpload.isErrorState) {
-    itemNode.classList.remove("file-list-item--progress");
+    itemNode.classList.remove(
+      "file-list-item--progress",
+      "file-list-item--loading",
+    );
     itemNode.classList.add("file-list-item--error");
   } else if (shouldShowAspectRatioWarning) {
     itemNode.classList.add("file-list-item--warning");
   }
+
+  if (imageUpload.isGeneratingPreviewState) {
+    setTimeout(() => {
+      itemNode.classList.remove("file-list-item--done");
+      itemNode.classList.add("file-list-item--loading");
+    }, 10);
```

Comment:
Why need setTimeout?

Reply:
seems did not remove progress bar. let's open issue to remove those unused code?